store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow.md

```diff
@@ -11,73 +11,346 @@ This tutorial will cover the end-to-end process of creating a bucket, uploading
+--8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/fileOperations.ts:imports'
+
+  // Add helper methods here
 
-```ts title="Create a Bucket"
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/create-a-bucket.ts'
 ```
 
-## Issue a Storage Request
+To implement the `uploadFile` helper method, add the following code to the `fileOperations.ts` file:
+
+```ts title="operations/fileOperations.ts // Add helper methods here"
+--8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/fileOperations.ts:upload-file-helper'
+```
 
-Ensure your file is ready to upload. In this demonstration, we're using a `.jpg` file named `hello.jpg` stored in the current working directory, i.e., the same as the typescript project files, `/src/`. 
+After a successful file upload the logs should look something like:
 
-Register your intent to store a file in your bucket and set its replication policy. Initialize `FileManager`, compute the file’s fingerprint, fetch MSP info (and extract peer IDs), choose a replication level and replica count, then call `issueStorageRequest`.
+--8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/upload-a-file/output-05.html'
 
-```ts title="Issue a Storage Request"
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/issue-storage-request.ts'
+### Call the Upload File Helper Method
+
+Replace the placeholder `// **PLACEHOLDER FOR STEP 4: UPLOAD FILE**` with the following code:
+
+```ts title="src/index.ts // **PLACEHOLDER FOR STEP 4: UPLOAD FILE**"
+  --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:upload-file'
 ```
 
-## Verify If Storage Request Is On-Chain
+??? code "View complete `index.ts` up until this point"
+
+    ```ts title="src/index.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:imports'
+
+    async function run() {
+    // For anything from @storagehub-sdk/core to work, initWasm() is required
+    // on top of the file
+    await initWasm();
+    
+    // --- End-to-end storage flow ---
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:check-msp-health'
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:create-bucket'
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:verify-bucket'
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:upload-file'
+    // **PLACEHOLDER FOR STEP 5: DOWNLOAD FILE**
+    // **PLACEHOLDER FOR STEP 6: VERIFY FILE**
+
+    // Disconnect the Polkadot API at the very end
+    await polkadotApi.disconnect();
+    }
+
+    await run();
```

Comment:
indentation needs to be fixed. can you pls double check all the snippets, I might have missed other ones that also have same issue